service/storage/encoding.go

```diff
@@ -25,6 +25,12 @@ func EncodeKey(prefix uint8, segments ...interface{}) []byte {
 		case flow.StateCommitment:
 			val = make([]byte, 32)
 			copy(val, s[:])
+		case flow.RegisterID:
+			val = make([]byte, 0, len(s.Owner)+1+len(s.Key)+1)
+			val = append(val, s.Owner...)
+			val = append(val, '/')
+			val = append(val, s.Key...)
+			val = append(val, '/')
```

Comment:
can we combine the code with [the lookupKey here](https://github.com/onflow/flow-archive/pull/104/files#diff-ec5ed77af8eef26b4b6e9d31f8ee434f8d53cf639d0bd9f7cc779e8fe7401435R24-R27)

Reply:
This code will go away after we switch payloads to the storage2 API.  storage depending on APIs from storage2 is a leaking abstraction it only makes sense if we factor this code out into a separate package, but that package would not make sense when we remove storage1 payloads API later

tl;dr is that case statement is implementation detail and would go away.